cinder/volume/api.py

```diff
@@ -2143,6 +2143,11 @@ def thaw_host(self, ctxt, host, cluster_name):
         if not self.volume_rpcapi.thaw_host(ctxt, services[0]):
             return "Backend reported error during thaw_host operation."
 
+    def recount_host_volumes(self, ctxt, host):
+        ctxt.authorize(svr_policy.RECOUNT_VOLUMES_POLICY)
+        ctxt = ctxt if ctxt.is_admin else ctxt.elevated()
```

Comment:
then the context is already elevated when it reaches `_count_host_stats()` later on and we don't need to elevate it again? or is the elevation just for good measure in case the method was called a different way?

Reply:
frankly I really don't know.   The other admin only based actions in this file are done this way, so that's why I'm just following their lead, such as the replication API calls for freeze_host, thaw_host.

Reply:
Calling `elevated()` on an admin or already elevated context probably (?) doesn't hurt, I guess.

Reply:
Talking about https://github.com/sapcc/cinder/blob/0e92f32d54b954d9c18bc8a45a22db4108561203/cinder/volume/manager.py#L520 in case that wasn't clear.

Reply:
yah that's what I thought as well.   The other admin command for replication host modifications do the same thing for whatever reason.